self-tests.yml

```diff
@@ -797,7 +799,7 @@ buildvariants:
       goos: linux
       goarch: amd64
       nodebin: /opt/node/bin
-      GOROOT: /opt/golang/go1.20
+      GOROOT: /opt/golang/go1.21
```

Comment:
We can't use go1.21 in our tests because Evergreen is not allowed to compile for go 1.21 yet (DEVPROD-3611).